dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java

```diff
@@ -0,0 +1,75 @@
+package com.dotcms.cms.login;
+
+import com.dotcms.filters.interceptor.Result;
+import com.dotcms.filters.interceptor.WebInterceptor;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.SecurityLogger;
+import com.liferay.portal.model.User;
+import com.liferay.portal.util.PortalUtil;
+import com.liferay.util.StringPool;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * Intercepts performs the logout at the end of the call in order to give chance
+ * to thirdparties implementation to patch the logout process, if needed
+ */
+public class LogoutWebInterceptor implements WebInterceptor{
+
+    private static final String      API_CALL                = "/dotAdmin/logout";
+    private static final String      API_SHOW                = "/dotAdmin/show-logout";
+
+
+    @Override
+    public String[] getFilters() {
+        return new String[] {
+                API_CALL + "*",API_SHOW + "*",
+        };
+    }
+
+    @Override
+    public Result intercept(final HttpServletRequest request, final HttpServletResponse response) {
+
+        Logger.info(this.getClass(), "Starting Logout --> " + request.getRequestURI());
+
+        try {
+
+            if (request.getRequestURI().contains(API_CALL)) {
+                Logger.info(this, () -> "Doing the logout");
+                final User user = PortalUtil.getUser(request);
+
+                APILocator.getLoginServiceAPI().doActionLogout(request, response);
+
+                if (null != user) {
+                    SecurityLogger.logInfo(this.getClass(), "User " +
+                            user.getFullName() + " (" + user.getUserId() + ") has logged out from IP: " + request.getRemoteAddr());
+                }
+            }
+
+            //response.sendRedirect(Config.getStringProperty("logout.url", "/dotAdmin/#/public/logout"));
+            final RequestDispatcher dispatcher = Config.CONTEXT
+                    .getRequestDispatcher("/html/portal/show-logout.jsp");
+            dispatcher.forward(request, response);
+            Logger.info(this, ()-> "Logout DONE");
+        }  catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6263950)